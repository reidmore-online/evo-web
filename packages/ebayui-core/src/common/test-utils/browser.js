import { fireEvent, createEvent } from "@marko/testing-library";
/**
 * Simulates a touch based scroll event over 4 animation frames.
 *
 * @param {HTMLElement} el The element to scroll.
 * @param {number} to The new scrollLeft for the element.
 * @param {function} cb A callback to call after the scroll.
 */
function simulateScroll(el, to, cb) {
    fireEvent.scroll(el);
    el.scrollLeft = to;
    setTimeout(cb, 600);
}

function waitFrames(count, cb) {
    if (count) {
        return requestAnimationFrame(() => {
            waitFrames(count - 1, cb);
        });
    }

    cb();
}

async function pressKey(el, info) {
    for (const event of [
        createEvent.keyDown(el, info),
        createEvent.keyUp(el, info),
    ]) {
        // we assign properties to them for older browsers (chrome 49)
        Object.keys(info).forEach((key) => {
            if (event[key] !== info[key]) {
                Object.defineProperty(event, key, { value: info[key] });
            }
        });

        await fireEvent(el, event);
    }
}

const fastAnimations = {
    // Adds an style to the document which forces all transitions to run more quickly for the tests.
    start(disable) {
        if (this.fastAnimationStyle) {
            return;
        }
        const time = disable ? 0 : 1;

        this.fastAnimationStyle = document.createElement("style");
        this.fastAnimationStyle.innerHTML = `*,details::details-content,*::before,*::after {
                animation-duration: ${time}ms !important;
                transition-duration: ${time}ms !important;
                transition-delay: ${time}ms !important;
                scroll-behavior: auto !important;
            }`;
        document.head.appendChild(this.fastAnimationStyle);
    },
    stop() {
        if (!this.fastAnimationStyle) {
            return;
        }

        document.head.removeChild(this.fastAnimationStyle);
        this.fastAnimationStyle = undefined;
    },
};

export { fastAnimations, waitFrames, pressKey, simulateScroll };
