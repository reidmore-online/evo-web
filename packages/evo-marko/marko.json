{
  "taglib-id": "evo-marko",
  "tags-dir": "./src/tags",
  "taglib-imports": [
    "./src/tags/evo-icon/marko.json",
    "./src/tags/evo-skeleton/marko.json"
  ]
}
