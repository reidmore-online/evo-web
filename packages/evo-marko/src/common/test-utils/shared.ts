export function getNItems(n: number, getAttrs: (i: number) => any[]) {
  return Array.from({ length: n }).map((_, i) => getAttrs(i));
}

// Helper method to render pagination items
export function getPaginationItems(
  n: number,
  href: boolean = false,
  selected: number = -1,
  navDisabled: boolean = false,
  skipPrevNext: boolean = false,
) {
  const items = [];
  if (!skipPrevNext) {
    items.push({
      type: "previous",
      href: href ? "#" : null,
      disabled: navDisabled,
    });
  }
  for (let i = 0; i < n; i++) {
    items.push({
      href: href ? "#" : null,
      current: i === selected,
    });
  }
  if (!skipPrevNext) {
    items.push({
      type: "next",
      href: href ? "#" : null,
      disabled: navDisabled,
    });
  }
  return items;
}
