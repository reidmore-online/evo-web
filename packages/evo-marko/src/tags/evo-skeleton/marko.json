{
  "taglib-id": "evo-marko-skeleton",
  "tags-dir": "./tags"
}
