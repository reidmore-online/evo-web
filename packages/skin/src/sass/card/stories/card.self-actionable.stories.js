export default { title: "Skin/Card/Self Actionable" };

export const verticalButton = () => `
<button class="card">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            Professional-grade rookie cards featuring future Hall of Famers in their awkward rookie photo phase.
        </div>
    </div>
</button>
`;

export const verticalLink = () => `
<a class="card" href="https://ebay.com">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            A collection proving that somewhere between a curveball, a three-pointer, and a slap shot lies the secret to happiness.
        </div>
    </div>
</a>
`;

export const horizontalButton = () => `
<button class="card card--horizontal">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            Professional-grade rookie cards featuring future Hall of Famers in their awkward rookie photo phase.
        </div>
    </div>
</button>
`;

export const horizontalLink = () => `
<a class="card card--horizontal" href="https://ebay.com">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            A collection proving that somewhere between a curveball, a three-pointer, and a slap shot lies the secret to happiness.
        </div>
    </div>
</a>
`;
