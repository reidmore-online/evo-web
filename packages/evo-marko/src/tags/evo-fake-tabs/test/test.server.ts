import { describe, it } from "vitest";
import { composeStories } from "@storybook/marko";
import { snapshotHTML } from "../../../common/test-utils/snapshots";
import * as stories from "../fake-tabs.stories"; // import all stories from the stories file
const { Default, NoPanel } = composeStories(stories);

describe("fake-tabs", () => {
  it("renders default", async () => {
    await snapshotHTML(Default);
  });

  it("renders no panel", async () => {
    await snapshotHTML(NoPanel);
  });

  it("renders with tabMatchesCurrentUrl=false", async () => {
    await snapshotHTML(Default, {tabMatchesCurrentUrl: false});
  });

  it("renders with other selected index", async () => {
    await snapshotHTML(Default, {selectedIndex: 2});
  });
  it("renders with no selected index", async () => {
    await snapshotHTML(Default, {selectedIndex: -1});
  });

});
