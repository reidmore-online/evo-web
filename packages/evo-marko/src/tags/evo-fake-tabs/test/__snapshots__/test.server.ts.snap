// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fake-tabs > renders default 1`] = `
<DocumentFragment>
  <div
    class="fake-tabs"
  >
    <ul
      class="fake-tabs__items"
    >
      <li
        class="fake-tabs__item"
      >
        <a>
          Tab 1
        </a>
      </li>
      <li
        class="fake-tabs__item"
      >
        <a
          aria-current="page"
        >
          Tab 2
        </a>
      </li>
      <li
        class="fake-tabs__item"
      >
        <a>
          Tab 3
        </a>
      </li>
    </ul>
    <div
      class="fake-tabs__content"
    >
      <div
        class="fake-tabs__panel"
      >
        <div
          class="fake-tabs__cell"
        >
          <div>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla ornare, quam at lacinia pretium, lacus urna luctus nisi, eget molestie massa tortor id lacus. Aenean ac fringilla lacus. Fusce vel dui ex. Vivamus luctus egestas nulla, non hendrerit purus luctus at. Maecenas vel diam enim. Pellentesque quam neque, porttitor tincidunt vestibulum at, dapibus sit amet tortor.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`fake-tabs > renders no panel 1`] = `
<DocumentFragment>
  <div
    class="fake-tabs"
  >
    <ul
      class="fake-tabs__items"
    >
      <li
        class="fake-tabs__item"
      >
        <a
          aria-current="page"
        >
          Tab 1
        </a>
      </li>
      <li
        class="fake-tabs__item"
      >
        <a>
          Tab 2
        </a>
      </li>
      <li
        class="fake-tabs__item"
      >
        <a>
          Tab 3
        </a>
      </li>
    </ul>
    <div
      class="fake-tabs__content"
    >
      <div
        class="fake-tabs__panel"
      >
        <div
          class="fake-tabs__cell"
        >
          <div />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`fake-tabs > renders with no selected index 1`] = `
<DocumentFragment>
  <div
    class="fake-tabs"
  >
    <ul
      class="fake-tabs__items"
    >
      <li
        class="fake-tabs__item"
      >
        <a>
          Tab 1
        </a>
      </li>
      <li
        class="fake-tabs__item"
      >
        <a>
          Tab 2
        </a>
      </li>
      <li
        class="fake-tabs__item"
      >
        <a>
          Tab 3
        </a>
      </li>
    </ul>
    <div
      class="fake-tabs__content"
    >
      <div
        class="fake-tabs__panel"
      >
        <div
          class="fake-tabs__cell"
        >
          <div>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla ornare, quam at lacinia pretium, lacus urna luctus nisi, eget molestie massa tortor id lacus. Aenean ac fringilla lacus. Fusce vel dui ex. Vivamus luctus egestas nulla, non hendrerit purus luctus at. Maecenas vel diam enim. Pellentesque quam neque, porttitor tincidunt vestibulum at, dapibus sit amet tortor.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`fake-tabs > renders with other selected index 1`] = `
<DocumentFragment>
  <div
    class="fake-tabs"
  >
    <ul
      class="fake-tabs__items"
    >
      <li
        class="fake-tabs__item"
      >
        <a>
          Tab 1
        </a>
      </li>
      <li
        class="fake-tabs__item"
      >
        <a>
          Tab 2
        </a>
      </li>
      <li
        class="fake-tabs__item"
      >
        <a
          aria-current="page"
        >
          Tab 3
        </a>
      </li>
    </ul>
    <div
      class="fake-tabs__content"
    >
      <div
        class="fake-tabs__panel"
      >
        <div
          class="fake-tabs__cell"
        >
          <div>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla ornare, quam at lacinia pretium, lacus urna luctus nisi, eget molestie massa tortor id lacus. Aenean ac fringilla lacus. Fusce vel dui ex. Vivamus luctus egestas nulla, non hendrerit purus luctus at. Maecenas vel diam enim. Pellentesque quam neque, porttitor tincidunt vestibulum at, dapibus sit amet tortor.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`fake-tabs > renders with tabMatchesCurrentUrl=false 1`] = `
<DocumentFragment>
  <div
    class="fake-tabs"
  >
    <ul
      class="fake-tabs__items"
    >
      <li
        class="fake-tabs__item"
      >
        <a>
          Tab 1
        </a>
      </li>
      <li
        class="fake-tabs__item"
      >
        <a
          aria-current="true"
        >
          Tab 2
        </a>
      </li>
      <li
        class="fake-tabs__item"
      >
        <a>
          Tab 3
        </a>
      </li>
    </ul>
    <div
      class="fake-tabs__content"
    >
      <div
        class="fake-tabs__panel"
      >
        <div
          class="fake-tabs__cell"
        >
          <div>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla ornare, quam at lacinia pretium, lacus urna luctus nisi, eget molestie massa tortor id lacus. Aenean ac fringilla lacus. Fusce vel dui ex. Vivamus luctus egestas nulla, non hendrerit purus luctus at. Maecenas vel diam enim. Pellentesque quam neque, porttitor tincidunt vestibulum at, dapibus sit amet tortor.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
